core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,457 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing data related to blog posts."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import python_utils
+import utils
+(base_models, user_models) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class BlogPostModel(base_models.BaseModel):
+    """Model to store blog post data.
+
+    The id of instances of this class is in the form of random hash of
+    12 chars.
+    """
+
+    # We use the model id as a key in the Takeout dict.
+    ID_IS_USED_AS_TAKEOUT_KEY = True
+
+    # The ID of the user the blog post is authored by.
+    author_id = datastore_services.StringProperty(required=True)
+    # Title of the blog post.
+    title = datastore_services.StringProperty(indexed=True, required=True)
+    # Content of the blog post.
+    content = datastore_services.TextProperty(indexed=False, required=True)
+    # The unique url fragment of the blog post. If the user directly enters the
```

Comment:
Will we autogenerate this based on the title, sort of like Medium does? (I think they hyphenate the words and stick a random ID on the end)

Reply:
Yeah, we will autogenerate this based on title, the random ID on the end will be the blog post id (as we have in topics).